js/ui/grid_core/ui.grid_core.column_headers.js

```diff
@@ -96,17 +96,34 @@ export const columnHeadersModule = {
                     const that = this;
 
                     return function($container, options) {
-                        const $content = column.command ? $container : createCellContent(that, $container, options);
-                        const caption = column.command !== 'expand' && column.caption;
+                        const caption = column.command === 'empty' ? that._getEmptyHeaderText() : column.caption;
+                        const needCellContent = !column.command || (caption && column.command !== 'expand');
+
+                        if(needCellContent) {
+                            const $content = createCellContent(that, $container, options);
 
-                        if(caption) {
                             $content.text(caption);
                         } else if(column.command) {
                             $container.html('&nbsp;');
                         }
                     };
                 },
 
+                _getEmptyHeaderText: function() {
+                    const hasHiddenColumns = this.component.getView('columnChooserView').hasHiddenColumns();
+                    const hasGroupedColumns = this.component.getView('headerPanel').hasGroupedColumns();
+
+                    if(hasHiddenColumns && hasGroupedColumns) {
+                        return messageLocalization.format('dxDataGrid-emptyHeaderWithColumnChooserAndGroupPanelText');
+                    } else if(hasGroupedColumns) {
+                        return messageLocalization.format('dxDataGrid-emptyHeaderWithGroupPanelText');
+                    } else if(hasHiddenColumns) {
+                        return messageLocalization.format('dxDataGrid-emptyHeaderWithColumnChooserText');
+                    }
+
+                    return '';
```

Comment:
Minor comment.

We can use more compact ```switch``` here, 
but it's a matter of taste so it's very minor comment :)

```suggestion
                    switch(true) {
                      case hasHiddenColumns && hasGroupedColumns:
                        return messageLocalization.format('dxDataGrid-emptyHeaderWithColumnChooserAndGroupPanelText');
                      case hasGroupedColumns:
                        return messageLocalization.format('dxDataGrid-emptyHeaderWithGroupPanelText');
                       case hasHiddenColumns:
                         return messageLocalization.format('dxDataGrid-emptyHeaderWithColumnChooserText');
                       default:
                         return '';
                    }
```